test/jdk/java/awt/print/PrinterJob/PageRanges.java

```diff
@@ -21,58 +21,65 @@
  * questions.
  */
 
-/**
+/*
  * @test
  * @bug 6575331
  * @key printer
  * @summary The specified pages should be printed.
- * @run main/manual=yesno PageRanges
+ * @library /java/awt/regtesthelpers
+ * @library /test/lib
+ * @build PassFailJFrame
+ * @build jtreg.SkippedException
+ * @run main/manual PageRanges
  */
 
-import java.awt.*;
-import java.awt.print.*;
+import java.awt.Graphics;
+import java.awt.print.PageFormat;
+import java.awt.print.Printable;
+import java.awt.print.PrinterException;
+import java.awt.print.PrinterJob;
+import jtreg.SkippedException;
 
 public class PageRanges implements Printable {
-
-    static String[] instr = {
-     "This test prints two jobs, and tests that the specified range",
-     "of pages is printed. You must have a printer installed for this test.",
-     "In the first dialog, select a page range of 2 to 3, and press OK",
-     "In the second dialog, select ALL, to print all pages (in total 5 pages).",
-     "Collect the two print outs and confirm the jobs printed correctly",
-    };
+    private static final String INSTRUCTIONS = """
+                 This test prints two jobs, and tests that the specified range.
+                 of pages is printed. You must have a printer installed for this test.
+                 In the first dialog, select a page range of 2 to 3, and press OK
+                 In the second dialog, select ALL, to print all pages (in total 5 pages).
+                 Collect the two print outs and confirm the jobs printed correctly.
```

Comment:
```suggestion
                 This test prints two jobs and tests that the specified range
                 of pages is printed.
                 In the first dialog, select a page range of 2 to 3, and press OK.
                 In the second dialog, select ALL, to print all pages (in total 5 pages).
                 Collect the two print outs and confirm the jobs are printed correctly.
```

Now the test won't start if there's no printer.